tests/test_nginx.py

```diff
@@ -10,6 +11,11 @@ def test_nginx_welcome_page(auto_container, host):
     """test that the default welcome page is served by the container."""
     host_port = auto_container.forwarded_ports[0].host_port
 
-    assert "Welcome to nginx" in host.check_output(
-        f"curl -sf --retry 5 --retry-connrefused http://localhost:{host_port}/"
-    )
+    # Retry 5 times with quadratic backoff delay
+    retries = 5
+    for i in range(retries):
+        resp = requests.get(f"http://localhost:{host_port}/", timeout=30)
```

Comment:
this doesn't work, as get is throwing a requests.exceptions.ConnectionError

Reply:
Indeed, so we do need an exception handler. I wanted to avoid that